Updating the README with more descriptive detail (#43)
o3de/sig-core: README.md

Meeting notes from the [O3DE Meetings](https://o3de.github.io/sig-core/meetings/).

# General Resources

# Open 3D Engine Special Interest Group for Core Systems (SIG Core)

This repository holds materials for the Open 3D Engine Foundation Special Interest Group for Core Systems (SIG-Core).

## Description of SIG Core

> We manage the Core Systems needed by O3DE libraries to fulfill the engines requirements. Primarily low-level systems such as memory management, application startup and file input/output are implemented by. See the [SIG's charter](governance/SIG%20Core%20Charter.md) for full details.

Reviewer: ```suggestion
We manage the Core Systems needed by O3DE and its libraries to fulfill the engines requirements. The SIG primarily owns low-level systems such as memory management, application startup and file input/output. See the [SIG's charter](governance/SIG%20Core%20Charter.md) for full details.
```

## What the SIG does:

> * Contributes to changes to O3DE Core Systems in library(AzCore/AzFramework/AzToolsFramework/AzGameFramework).  

Reviewer: Doesn't the SIG owns most of AzCore/AzFramework? It seems odd just to call out AzGameFramework IMHO

Author: Yeah it is quite specific.
I'll remove it from the list of Core Systems
